airflow/contrib/hooks/aws_hook.py

```diff
@@ -156,6 +156,9 @@ def _get_credentials(self, region_name):
                     aws_session_token = credentials['SessionToken']
 
                 endpoint_url = extra_config.get('host')
+                # If there is an external session token use this token
```

Comment:
This needs to be done _before_ the assume_role call (as that will change the key and access token, and create it's _own_ session token) . And we also need to pass in the token to the `sts_session` we create on line 139